File: app/src/main/java/com/haishinkit/media/IEncoderListener.java
package com.haishinkit.media;

import java.nio.ByteBuffer;
import android.media.MediaCodec.BufferInfo;
import android.media.MediaCodecInfo;
import android.media.MediaFormat;

public interface IEncoderListener {
    public void onFormatChanged(final String mime, final MediaFormat mediaFormat);
    public void onSampleOutput(final String mime, final BufferInfo info, final ByteBuffer buffer);
}

File: haishinkit/src/main/java/com/haishinkit/media/codec/IEncoderListener.kt
package com.haishinkit.media.codec

import java.nio.ByteBuffer

import android.media.MediaCodec.BufferInfo
import android.media.MediaFormat

interface IEncoderListener {
    fun onFormatChanged(mime: String, mediaFormat: MediaFormat)
    fun onSampleOutput(mime: String, info: BufferInfo, buffer: ByteBuffer)
}
